mirdata/core.py

```diff
@@ -271,6 +278,49 @@ class Track(object):
 
     """
 
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata=None,
+    ):
+        """Track init method. Sets boilerplate attributes, including:
+
+        - ``track_id``
+        - ``_dataset_name``
+        - ``_data_home``
+        - ``_track_paths``
+        - ``_track_metadata``
+
+        Args:
+            track_id (str): track id
+            data_home (str): path where mirdata will look for the dataset
+            dataset_name (str): the identifier of the dataset
+            index (dict): the dataset's file index
+                Typically accessed via the .index attribute of a LargeData object
+            metadata (dict or None): a dictionary of metadata or None
+
+        """
+        if track_id not in index["tracks"]:
+            raise ValueError(
+                "{} is not a valid track_id in {}".format(track_id, dataset_name)
+            )
+
+        self.track_id = track_id
+        self._dataset_name = dataset_name
+
+        self._data_home = data_home
+        self._track_paths = index["tracks"][track_id]
+
+        if metadata and track_id in metadata:
+            self._track_metadata = metadata[track_id]
+        elif metadata:
+            self._track_metadata = metadata
+        else:
+            self._track_metadata = {}
```

Comment:
good point - changing it! Worth mentioning - remember that before we let missing metadata fail gracefully, because the loaders wouldn't initialize if it wasn't yet downloaded? Now it's not so much of a problem, and I changed it to raise an error if the metadata is accessed but the file isn't there.